### DISCUSSION TITLE
Improve UX for hiding NULLs in individual columns

### DISCUSSION THREAD

**Discussion by NickCrews** [directus/directus#20738] [Feature Requests]
### Summary

Started from https://github.com/directus/directus/issues/20723

Currently, we use the default sort order: [NULLS FIRST is the default for DESC order, and NULLS LAST otherwise](https://www.postgresql.org/docs/current/queries-order.html). Thus if I sort DESC the first page of results is all NULLs.

But, at least for me, at least 90% of the time when I am sorting like this I don't want to see NULLs. eg for date rows, I want to see the earliest or the latest rows. For prices I want to see the mins or the maxes.

But, as mentioned in the issue above, we also want to avoid the footgun of people not noticing that their column has NULLs. We want it to still be obvious if NULLs are present. So always sorting NULLs last wouldn't be great.

At this point, to solve those two requirements, my first thought would be to just increase the options of sorting from just "asc" and "desc" to:

- asc, nulls last
- asc, nulls first
- desc, nulls last
- desc, nulls first

Or, we could tackle this from a different angle, using filters:  Add a "hide NULLs' toggle to each column, in the same sorting modal? I wouldn't want that toggle to get hidden and then forgotten about, so I accidentally miss rows later. The UX could be that clicking the checkbox just goes up to the search bar and adds a `AND column NOT NULL` clause? This would be hard to undo, eg I click the checkbox, then manually add some other filter, then I unclick the checkbox, how to find the right clause to remove from the filter? Maybe there is just a button that adds the clause, you are responsible to removing the clause?

### Basic Example

the sorting modal now has 4 options, not the current 2.

### Motivation

See above

### Detailed Design

see above

### Must Have:
- hide NULLs when I don't want to see them. Note I am phrasing it like this, because this could be accomplished using NULLS LAST sort order, or using a NOT NULL filter.
- make the presence of NULLs obvious

### Should Have:
- C

### Could Have:
- different default behavior for frontend usage.

### Won't Have:
- different default behavio for API usage.

### Drawbacks

- I think this is pretty explainable to users
- should be a pretty self-contained if we follow the ORDER BY route, just a few new conditionals.

### Alternatives

just add the NOT NULL filter manually to the filter box

### Adoption Strategy

need to hash out more details before we can talk about this I think

**Comment by sheikhWasib** [comment]
many people have question how we download nulls brawl apk some trusted source. so use null brawl and I get a trusted source if you want to download here is link [[Nulls brawl](https://apknullsbrawl.com/)]apk indir

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
